modules/aniviewBidAdapter.js

```diff
@@ -0,0 +1,191 @@
+    let eventCallback = bid && bid.renderer && bid.renderer.handleVideoEvent ? bid.renderer.handleVideoEvent : null;
+    window.aniviewRenderer.renderAd({
+      id: bid.adUnitCode + '_' + bid.adId,
+      debug: window.location.href.indexOf('pbjsDebug') >= 0,
+      placement: bid.adUnitCode,
+      width: bid.width,
+      height: bid.height,
+      vastUrl: bid.vastUrl,
+      vastXml: bid.vastXml,
+      config: bid.params[0].rendererConfig,
+      eventsCallback: eventCallback,
+      bid: bid
+    });
+  });
+}
+
+function newRenderer(bidRequest) {
+  const renderer = Renderer.install({
+    url: 'https://player.aniview.com/script/6.1/prebidRenderer.js',
+    config: {},
+    loaded: false,
+  });
+
+  try {
+    renderer.setRender(avRenderer);
+  } catch (err) {
+  }
+
+  return renderer;
+}
+
+function isBidRequestValid(bid) {
+  if (!bid.params || !bid.params.AV_PUBLISHERID || !bid.params.AV_CHANNELID) { return false; }
+
+  return true;
+}
+
+function buildRequests(validBidRequests, bidderRequest) {
+  let bidRequests = [];
+
+  for (let i = 0; i < validBidRequests.length; i++) {
+    let bidRequest = validBidRequests[i];
+
+    if (!bidRequest.sizes || !bidRequest.sizes.length) {
+      bidRequest.sizes = [[640, 480]];
+    }
+
+    if (bidRequest.sizes.length === 2 && typeof bidRequest.sizes[0] === 'number' && typeof bidRequest.sizes[1] === 'number') {
+      let adWidth = bidRequest.sizes[0];
+      let adHeight = bidRequest.sizes[1];
+      bidRequest.sizes = [[adWidth, adHeight]];
+    }
+
+    for (let j = 0; j < bidRequest.sizes.length; j++) {
+      let size = bidRequest.sizes[j];
+      let playerWidth;
+      let playerHeight;
+      if (size && size.length == 2) {
+        playerWidth = size[0];
+        playerHeight = size[1];
+      } else {
+        playerWidth = 640;
+        playerHeight = 480;
+      }
+
+      let s2sParams = {};
+
+      for (var attrname in bidRequest.params) {
+        if (bidRequest.params.hasOwnProperty(attrname) && attrname.indexOf('AV_') == 0) {
+          s2sParams[attrname] = bidRequest.params[attrname];
+        }
+      };
+
+      if (s2sParams.AV_APPPKGNAME && !s2sParams.AV_URL) { s2sParams.AV_URL = s2sParams.AV_APPPKGNAME; }
+      if (!s2sParams.AV_IDFA && !s2sParams.AV_URL) { s2sParams.AV_URL = utils.getTopWindowUrl(); }
+      if (s2sParams.AV_IDFA && !s2sParams.AV_AID) { s2sParams.AV_AID = s2sParams.AV_IDFA; }
+      if (s2sParams.AV_AID && !s2sParams.AV_IDFA) { s2sParams.AV_IDFA = s2sParams.AV_AID; }
+
+      s2sParams.pbjs = 1;
+      s2sParams.cb = Math.floor(Math.random() * 999999999);
+      s2sParams.AV_WIDTH = playerWidth;
+      s2sParams.AV_HEIGHT = playerHeight;
+      s2sParams.s2s = '1';
+      s2sParams.bidId = bidRequest.bidId;
+      s2sParams.bidWidth = playerWidth;
+      s2sParams.bidHeight = playerHeight;
+
+      if (bidderRequest && bidderRequest.gdprConsent) {
+        if (bidderRequest.gdprConsent.gdprApplies) {
+          AV_GDPR = 1;
```

Comment:
GDPR data is being stored on the window object, is this intentional?
Here are the resulting window properties created from lines 100 and 101:
```
window.AV_GDPR = 1
window.AV_CONSENT = "..."
```